### PR TITLE
Migrate from gradle-enterprise to develocity plugin

### PR DESCRIPTION
**Pull request by jprinet**
Along with Develocity [2024.1](https://gradle.com/develocity/releases/2024.1), the plugin id has been migrated to `com.gradle:develocity-gradle-plugin`.
The previous API is still compatible but using it with the new plugin version leads to deprecation warnings.

This PR upgrades the plugin and adjust the configuration to the new `develocity` extension

### CODE REVIEW

**Comment by jprinet** [comment]
cc @CristianGM

File: repo/gradle-settings-conventions/develocity/src/main/kotlin/develocity.settings.gradle.kts [hunk 1]
```diff
@@ -11,21 +11,18 @@ if (buildProperties.buildScanServer != null) {
...
-                isBuildLogging = true
-                isBuildLogging = true
-                isUploadInBackground = true
+                buildLogging = true
+                uploadInBackground = true
             }
         } else {
-            termsOfServiceUrl = "https://gradle.com/terms-of-service"
-            termsOfServiceAgree = "yes"
+            termsOfUseUrl = "https://gradle.com/terms-of-service"
```
Comment (runningcode) [inline repo/gradle-settings-conventions/develocity/src/main/kotlin/develocity.settings.gradle.kts:24]:
```suggestion
            termsOfUseUrl = "https://gradle.com/help/legal-terms-of-use"
```

File: repo/gradle-settings-conventions/develocity/src/main/kotlin/develocity.settings.gradle.kts [hunk 2]
```diff
@@ -11,21 +11,18 @@ if (buildProperties.buildScanServer != null) {
...
             server = buildScanServer
-            publishAlways()
 
             capture {
-                isTaskInputFiles = true
-                isBuildLogging = true
-                isBuildLogging = true
-                isUploadInBackground = true
+                buildLogging = true
+                uploadInBackground = true
```
Comment (runningcode) [inline repo/gradle-settings-conventions/develocity/src/main/kotlin/develocity.settings.gradle.kts:21]: this should be conditioned on being a local build. but this is not a regression.

Comment (jprinet) [reply repo/gradle-settings-conventions/develocity/src/main/kotlin/develocity.settings.gradle.kts:21]: I leave it to the maintainers to decide whether to introduce this change here or not

**Comment by CristianGM** [comment]
Merged in https://github.com/JetBrains/kotlin/commit/ebc10095c0dc79790ab1d5d8f8709dcd77db4397, thank you!

**Pull request closed without merging**